server/api/src/rs_get_resource_info_for_operation.cpp

```diff
@@ -0,0 +1,85 @@
+#include "irods/rcGlobalExtern.h"
+#include "irods/getRemoteZoneResc.h"
+#include "irods/dataObjCreate.h"
+#include "irods/objMetaOpr.hpp"
+#include "irods/resource.hpp"
+#include "irods/collection.hpp"
+#include "irods/specColl.hpp"
+#include "irods/miscServerFunct.hpp"
+#include "irods/openCollection.h"
+#include "irods/readCollection.h"
+#include "irods/closeCollection.h"
+#include "irods/dataObjOpr.hpp"
+#include "irods/rsGetHostForGet.hpp"
+#include "irods/irods_resource_backport.hpp"
+#include "irods/irods_resource_redirect.hpp"
+#include "irods/irods_at_scope_exit.hpp"
+
+#include <nlohmann/json.hpp>
+
+#include <set>
+#include <cstring>      // for strdup
+#include <optional>
+
+int rs_get_resource_info_for_operation(
+    rsComm_t*     rsComm,
+    dataObjInp_t* dataObjInp,
+    char **       out_info )
+{
+    rodsServerHost_t *rodsServerHost;
+    const int remoteFlag = getAndConnRemoteZone(rsComm, dataObjInp, &rodsServerHost, REMOTE_OPEN);
+    if (remoteFlag < 0) {
+        return remoteFlag;
+    }
+    else if (REMOTE_HOST == remoteFlag) {
+        const int status = rc_get_resource_info_for_operation(rodsServerHost->conn, dataObjInp, out_info);
+        if (status < 0) {
+            return status;
+        }
+    }
+    else {
+        std::string location {};
+        // =-=-=-=-=-=-=-
+        // working on the "home zone", determine if we need to redirect to a different
+        // server in this zone for this operation.  if there is a RESC_HIER_STR_KW then
+        // we know that the redirection decision has already been made
+        std::string hier{};
+        const char * op_type = getValByKey(&dataObjInp->condInput, GET_RESOURCE_INFO_OP_TYPE_KW);
+        std::set<std::string> allowed_ops{ irods::CREATE_OPERATION,
+                                           irods::WRITE_OPERATION,
+                                           irods::OPEN_OPERATION  };
+        if (!op_type || std::find(allowed_ops.begin(), allowed_ops.end(), op_type) == allowed_ops.end()) {
+            return RE_UNSUPPORTED_OP_OR_TYPE;
+        }
+        const std::string operation{op_type};
+        if ( getValByKey( &dataObjInp->condInput, RESC_HIER_STR_KW ) == NULL ) {
+            try {
+                auto result = irods::resolve_resource_hierarchy(operation, rsComm, *dataObjInp);
+                hier = std::get<std::string>(result);
+            }
+            catch (const irods::exception& e ) {
+                irods::log(e);
+                return e.code();
+            }
+        } // if keyword
+
+        // =-=-=-=-=-=-=-
+        // extract the host location from the resource hierarchy
+        irods::error ret = irods::get_loc_for_hier_string(hier, location);
+        if ( !ret.ok() ) {
+            irods::log( PASSMSG( "rs_get_resource_info_for_operation - failed in get_loc_for_hier_string", ret ) );
+            return -1;
+        }
+
+        // =-=-=-=-=-=-=-
+        // set the out variable
+        nlohmann::json J;
```

Comment:
No uppercase variable names please.

Consider renaming so it helps the read understand its purpose. For example, this could be called `resc_info`.